### PR TITLE
Remove RAY_CHECK from JNI code

### PR DESCRIPTION
**Pull request by guoyuhong**
## What do these changes do?
When we are testing FO, we will kill raylet using "kill -9". We found there are many core dump files from JAVA. We'd better remove the code that causes JNI crash. Instead, we could throw Java exceptions. For one thing, it is not easy to debug core dump files. For another, JNI crash will crash the whole JVM program.

## Related issue number
N/A

### CODE REVIEW

**Review by robertnishihara — COMMENTED** [review]
This seems fine to me.

**Review by robertnishihara — APPROVED** [review]

File: java/test/src/main/java/org/ray/api/test/ClientExceptionTest.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        System.out.println("Got InterruptedException when sleeping, exit right now.");
+        throw new RuntimeException("Got InterruptedException when sleeping.", e);
+      }
+    });
+    thread.start();
+    try {
+      Ray.wait(ImmutableList.of(notExisting), 1, 2000);
+      Assert.fail("Should not reach here");
+    } catch (RayException e) {
+      System.out.println(String.format("Expected runtime exception: {}", e));
```
Comment (raulchen) [inline java/test/src/main/java/org/ray/api/test/ClientExceptionTest.java]: nit, better use `LOGGER.debug` here

File: java/test/src/main/java/org/ray/api/test/ClientExceptionTest.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    }
+    try {
+      thread.join();
+    } catch (Exception e) {
+      System.out.println(e);
```
Comment (raulchen) [inline java/test/src/main/java/org/ray/api/test/ClientExceptionTest.java]: `LOGGER.error` here

File: src/ray/raylet/lib/java/org_ray_runtime_raylet_RayletClientImpl.cc [hunk 3]
```diff
@@ -31,6 +31,17 @@ class UniqueIdFromJByteArray {
   }
 };
 
+inline void ThrowRayExceptionIfNotOK(JNIEnv *env, const ray::Status &status,
+                                     const std::string &message) {
+  if (!status.ok()) {
+    jclass exception_class = env->FindClass("org/ray/api/exception/RayException");
+    if (exception_class == 0) {
+      exception_class = env->FindClass("java/lang/RuntimeException");
```
Comment (raulchen) [inline src/ray/raylet/lib/java/org_ray_runtime_raylet_RayletClientImpl.cc]: why is this needed?

**Review by raulchen — CHANGES_REQUESTED** [review]
thanks, I left a few minor comments.

File: src/ray/raylet/lib/java/org_ray_runtime_raylet_RayletClientImpl.cc (hunk 3 above)
Comment (guoyuhong) [reply src/ray/raylet/lib/java/org_ray_runtime_raylet_RayletClientImpl.cc]: When I implemented this test at the first time, I used `.get()` but I still found there are core dump files. This code is used to debug to see whether the `RayException` class is found or not. I found the crash is caused by Plasma Client. I will later remove the `ARROW_CHECK` out of plasma client JNI. I will remove these 3 lines.

**Review by raulchen — APPROVED** [review]

**Comment by guoyuhong** [comment]
All test failures were python failures, which are not caused by this PR. I will merge this PR and try to solve the CI failures in another PR.

**Comment by zhijunfu** [comment]
this looks good. maybe we can consider changing the C++ side as well. right now there are checks in raylet and plasma client.

**Comment by guoyuhong** [comment]
@zhijunfu Yes, I have removed the `RAY_CHECK`s from redis module and JNI code. The python client now is in cython and it throws exceptions when error happens. Currently, we need to refine the Plasma Client.
